group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupTest.java

```diff
@@ -568,5 +568,20 @@ public void testMetadataRefreshDeadline() {
         assertTrue(group.hasMetadataExpired(time.milliseconds()));
         assertEquals(time.milliseconds() + 1000, group.metadataRefreshDeadline().deadlineMs);
         assertEquals(group.groupEpoch() + 1, group.metadataRefreshDeadline().epoch);
+
+        // Set the refresh deadline.
+        group.setMetadataRefreshDeadline(time.milliseconds() + 1000, group.groupEpoch());
```

Comment:
we can just go back an epoch?

Reply:
i don't understand your comment. could you elaborate?

Reply:
btw, i just fixed the previous test case at L566.

Reply:
this could for instance if the write is lost so the new epoch would not be known anymore.

Reply:
I thought in the case the write is lost, we have the higher epoch in the DeadlineAndEpoch and that would signal us to continue to refresh. However, in this case, you are saying we would go back and epoch and decide not to try to refresh anymore. Is that correct?

Reply:
Yeah, you’re right. I actually use the current group epoch here because hasMetadataExpired would be true if epoch + 1.

Reply:
Ok -- so this is behavior we want then? I guess I was just having trouble seeing when we would update to the lower epoch.

Reply:
I think so. The epoch value is not so important here. The important part is that it should be reset regardless.

Reply:
Hmm -- reset meaning we should refresh? I guess my point was that if we lower the epoch we may delay the reset. I guess worst case, we just have the wait the refresh interval though.

Reply:
I have reworked the test to reduce the confusion. Let me know what you think.